ci/code_checks.sh

```diff
@@ -33,56 +33,63 @@ function invgrep {
     #
     # This is useful for the CI, as we want to fail if one of the patterns
     # that we want to avoid is found by grep.
-    if [[ "$AZURE" == "true" ]]; then
+    if [[ "${AZURE}" == "true" ]]; then
         set -o pipefail
         grep -n "$@" | awk -F ":" '{print "##vso[task.logissue type=error;sourcepath=" $1 ";linenumber=" $2 ";] Found unwanted pattern: " $3}'
     else
         grep "$@"
     fi
-    return $((! $?))
+    retval=$?
+    return $((! "${retval}"))
 }
 
-if [[ "$AZURE" == "true" ]]; then
+if [[ "${AZURE}" == "true" ]]; then
     FLAKE8_FORMAT="##vso[task.logissue type=error;sourcepath=%(path)s;linenumber=%(row)s;columnnumber=%(col)s;code=%(code)s;]%(text)s"
 else
     FLAKE8_FORMAT="default"
 fi
 
 ### LINTING ###
-if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
+if [[ -z "${CHECK}" || "${CHECK}" == "lint" ]]; then
 
     # `setup.cfg` contains the list of error codes that are being ignored in flake8
 
-    echo "flake8 --version"
-    flake8 --version
+    # Check if flake8 is installed.
```

Comment:
you can do the check on flake8 w/o moving the code itself, eg. revert lines 62 and on.
if the command doesn't exist then need to print a message and immediately exit